runtime/parachains/src/hrmp.rs

```diff
@@ -458,8 +518,10 @@ pub mod pallet {
 		///
 		/// If there are pending HRMP open channel requests, you can use this
 		/// function process all of those requests immediately.
-		#[pallet::weight(0)]
-		pub fn force_process_hrmp_open(origin: OriginFor<T>) -> DispatchResult {
+		///
+		/// Total number of opening channels must be provided as witness data of weighing.
+		#[pallet::weight(<T as Config>::WeightInfo::force_process_hrmp_open(*_channels))]
```

Comment:
This appears to be not checked. Do you think this right? It also applies to other places above and below.

In any case, this looks weird for reviewers/auditors and as such required a comment.

Reply:
It is not checked because it is coming from root origin and it is assumed to be correct, but I could easily add that, but IMO it is just noise. It will clutter both the dispatch code and the error variant. 

About the comment, if you look around Frame, this a pretty common patter, where some additional data needs to be passed to the dispatch, sometime not really used other than checked for correctness, just to measure the weight of some transaction ahead of time and have efficient pool processing. These are typically called "witness data".